src/i18n.ts

```diff
@@ -0,0 +1,280 @@
+    },
+  ): void {
+    this.fluent.addTranslation({
+      locales: locale,
+      isDefault: locale === this.config.defaultLocale,
+      bundleOptions: this.config.fluentBundleOptions,
+      ...options,
+    });
+
+    this.locales.push(locale);
+  }
+
+  /** Gets an array of translations for a given key. */
+  t(key: string): Array<string>;
+  /** Gets a message by its key from the specified locale. */
+  t(locale: LocaleId, key: string, context?: TranslationContext): string;
+  t(
+    keyOrLocale: string | LocaleId,
+    key?: string,
+    context?: TranslationContext,
+  ): string | Array<string> {
+    if (keyOrLocale && key) {
+      return this.fluent.translate(keyOrLocale, key, context);
+    }
+
+    const translations = new Array<string>();
+    for (const locale of this.locales) {
+      const message = this.translate(locale, keyOrLocale, context);
+      translations.push(message);
+    }
+
+    return translations;
+  }
+
+  /** Gets a message by its key from the specified locale. */
+  translate(key: string): Array<string>;
+  translate(
+    locale: LocaleId,
+    key: string,
+    context?: TranslationContext,
+  ): string;
+  translate(
+    keyOrLocale: string,
+    key?: string,
+    context?: TranslationContext,
+  ): string | string[] {
+    if (keyOrLocale && key) {
+      return this.fluent.translate(keyOrLocale, key, context);
+    }
+
+    const translations = new Array<string>();
+    for (const locale of this.locales) {
+      const message = this.translate(locale, keyOrLocale, context);
+      translations.push(message);
+    }
+
+    return translations;
+  }
+
+  /** Returns a middleware to .use on the `Bot` instance. */
+  middleware(): Middleware<C & I18nContextFlavor> {
+    const fluent = this.fluent;
+    const { defaultLocale, localeNegotiator, useSession } = this.config;
+    return async function (
+      ctx: C & I18nContextFlavor,
+      next: NextFunction,
+    ): Promise<void> {
+      let translate: TranslateFunction;
+
+      ctx.i18n = {
+        fluent,
+        renegotiateLocale: negotiateLocale,
+        useLocale,
+        getLocale: getNegotiatedLocale,
+        setLocale,
+      };
+      ctx.t = translateWrapper;
+      ctx.translate = translateWrapper;
+
+      await negotiateLocale();
+      await next();
+
+      // Also exports ctx object properties for accessing them directly from
+      // the translation source files.
+      function translateWrapper(
+        key: string,
+        context?: TranslationContext,
+      ): string {
+        return translate(key, {
+          first_name: ctx.from?.first_name ?? "",
+          ctx: JSON.stringify(makeContextObject(ctx)),
+          ...context,
+        });
+      }
+
+      async function getNegotiatedLocale(): Promise<string> {
+        return await localeNegotiator?.(ctx) ??
+          // deno-lint-ignore no-explicit-any
+          (await (useSession && (ctx as any).session))?.__language_code ??
```

Comment:
I have my concern in here because the code is short and I am lost 😆. On the other hand, we have `any` keyword. Do you think it's possible to set type for `ctx`?

Reply:
The `any` is fine there imho.

Reply:
I'm be honest that any could make the code looks least scary but I just share my opinion from a guy just join doesn't understand where the session coming from. Since dcdunkan made it clear of how, we can keep the `any` in here for now. I will come up a good name for it in the future. (yeah, we should ignore it for now)